pymc_bart/pgbart.py

```diff
@@ -597,3 +609,27 @@ def logp(point, out_vars, vars, shared):  # pylint: disable=redefined-builtin
     function = pytensor_function([inarray0], out_list[0])
     function.trust_input = True
     return function
+
+
+def fast_linear_fit():
+    """If available use Numba to speed up the computation of the linear fit"""
+
+    def linear_fit(X, Y):
+
+        n = len(Y)
+        xbar = np.sum(X) / n
+        ybar = np.sum(Y) / n
+
+        b = (X @ Y - n * xbar * ybar) / (X @ X - n * xbar**2)
+        a = ybar - b * xbar
+
+        y_fit = a + b * X
+        return y_fit, (a, b)
+
+    try:
+        from numba import jit
+
+        return jit(linear_fit)
+
+    except ImportError:
+        return linear_fit
```

Comment:
numba is a dependency, so just do this. Also move this up closer to `fast_mean`

```suggestion
def fast_linear_fit(X, Y):
    """Use Numba to speed up the computation of the linear fit"""
    n = len(Y)
    xbar = np.sum(X) / n
    ybar = np.sum(Y) / n

    b = (X @ Y - n * xbar * ybar) / (X @ X - n * xbar**2)
    a = ybar - b * xbar

    y_fit = a + b * X
    return y_fit, (a, b)
```